conmon-rs/server/src/child_reaper.rs

```diff
@@ -59,40 +60,77 @@ impl ChildReaper {
         Ok(())
     }
 
-    pub fn create_child<P, I, S>(&self, cmd: P, args: I) -> Result<std::process::ExitStatus>
+    pub async fn create_child<P, I, S>(&self, cmd: P, args: I) -> Result<ExitStatus>
     where
         P: AsRef<std::ffi::OsStr>,
         I: IntoIterator<Item = S>,
         S: AsRef<std::ffi::OsStr>,
     {
-        let mut cmd = std::process::Command::new(cmd);
+        let mut cmd = tokio::process::Command::new(cmd);
         cmd.args(args);
-
-        // To prevent an error: "No child processes (os error 10)",
-        // we prevent our reaper thread from calling waitpid until this child has completed.
-        // Inspired by
-        // https://github.com/kata-containers/kata-containers/blob/828a304883a4/src/agent/src/signal.rs#L27
-        let wait_lock = Arc::clone(&self.wait_lock);
-        let _lock = wait_lock
-            .lock()
-            .map_err(|e| format_err!("lock waitlock: {}", e))?;
-
         cmd.spawn()
             .map_err(|e| format_err!("spawn child process: {}", e))?
             .wait()
+            .await
             .map_err(|e| format_err!("wait for child process: {}", e))
     }
 
+    pub async fn exec_sync(
+        &self,
+        command: &Path,
+        args: Vec<String>,
+        timeout: i32,
+    ) -> Result<(ExitStatus, String, String)> {
+        let child = tokio::process::Command::new(command)
+            .stdout(Stdio::piped())
+            .stderr(Stdio::piped())
+            .kill_on_drop(true)
+            .args(args)
+            .spawn()
+            .map_err(|e| format_err!("spawn child process: {}", e))?;
+
+        let delay = tokio::time::sleep(tokio::time::Duration::from_secs(timeout as u64));
+        tokio::pin!(delay);
+        tokio::select! {
+            _ = &mut delay => {
+                debug!("timeout");
+                return Err(format_err!("timeout error"))
+            },
+            status = child.wait_with_output() => {
+                 let output = status.expect("blah");
+                 let stdout = String::from_utf8(output.stdout)?;
```

Comment:
I think we don't want to error out if the child output invalid UTF-8 somehow.  I'd use [from_utf8_lossy](https://doc.rust-lang.org/std/string/struct.String.html#method.from_utf8_lossy) instead.

(And what's better, we avoid duplicating the buffer in the happy path)

However, since this function returns owned values, we probably want to change it to return the raw `Vec<u8>`, and do the utf8 bits in the callers.